src/editor/esphome-editor.ts

```diff
@@ -70,6 +74,60 @@ class ESPHomeEditor extends LitElement {
           white-space: nowrap;
           text-overflow: ellipsis;
         }
+      </style>
+    `;
+    const isSecrets = this._isSecrets();
+    if (isSecrets && this.streamerMode && !this._showSecrets) {
```

Comment:
This feels weird to put it in this component. Instead, this dialog should be it’s own code that is called from the “Secrets” button. When user confirms, it will continue to open the editor.

Reply:
That could be an approach, but this will work for when I've implemented routing the secrets page will have it;s own url `/secrets` the button will be a link to the page not a change in state...

Reply:
We can always attach an event listener to the link to avoid it linking in streamer mode.

Reply:
I don't understand why you're not ok with this being in the editor.
There was already logic for the editor to handle secrets differently (no install & ace websocket not started)
so I don't see why it cannot be given 1 extra parameter (streamer mode) in order to show the warning 1st

Reply:
A component should do 1 thing. This makes it very complicated with doing completely different things. Also, it's a prompt to a user so we should follow the convention of asking that as a dialog.

Reply:
OK created a dialog in `streamer-warning.ts`
![image](https://github.com/esphome/dashboard/assets/771817/41bb4a35-9031-43d9-834e-da9ed9b4b710)

Reply:
@balloob are you happy with this change?